src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -82,24 +88,80 @@ private static String convertToJson(UserAuth userAuth) {
   
   /**
    * If the user is not in database already, add them.
+   * This method returns the Entity object in the database (or the newly-created
+   * user Entity, if none previously existed).
    */
-  public void addUserToDatabase(String email) {
+  public static Entity addUserToDatabase(String email) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Only add User to database if they are not already present there.
+    Entity userEntity = getUserEntityFromEmail(email);
+    if (userEntity == null) {
+      // User is not in database, create and add user object.
+      User newUser = new User(email);
+      Entity newUserEntity = newUser.buildEntity();
```

Comment:
unsure but do u test buildEntity in your User class?

Reply:
I don't; I test it in a future CL, but I'll move those changes downstream to this branch.

Reply:
Actually, nevermind -- I do test this!